src/re_frisk_remote/core.cljs

```diff
@@ -5,79 +5,165 @@
+                       {:event value})
+          app-db-diff (diff/diff prev-event-app-db app-db)
+          payload (conj {:app-db-diff app-db-diff}
+                        event-data)]
+      (swap! send-state assoc :prev-event-app-db app-db :event-time nil)
+      (send [:refrisk/events payload]))))
+
+(defn id-handlers []
+  (into {} (map #(hash-map (first %) @(second %))
+                (remove #(or (= (first (ffirst %)) ::db)
+                             (nil? (second %)))
+                        @query->reaction))))
+
+(defn id-handlers-msg []
+  (let [ih-prev (:prev-id-handlers @send-state)
+        ih (id-handlers)]
+    (swap! send-state assoc :prev-id-handlers ih)
+    (if ih-prev
+      (when-let [d (delta/delta ih-prev ih)]
+        [:refrisk/id-handler-delta d])
+      [:refrisk/id-handler ih])))
 
 (defn re-frame-sub [& rest]
-  ;; TODO send diff
-  (@chsk-send!* [:refrisk/id-handler
-                 (into {} (map #(hash-map (first %) @(second %))
-                               (remove #(or (= (first (ffirst %)) ::db)
-                                            (nil? (second %)))
-                                       @query->reaction)))])
+  (when @send-state
+    (send (id-handlers-msg)))
   (js/clearInterval @id-handler-timer*)
   (reset! id-handler-timer* nil))
 
+(defn- get-db []
+  (let [db @(subscribe [::db])]
+    (if @pre-send* (@pre-send* db) db)))
+
+(defn- app-db-msg [db]
+  (let [db-prev (:prev-app-db @send-state)]
+    (swap! send-state assoc :prev-app-db db)
+    (if db-prev
+      (when-let [d (delta/delta db-prev db)]
+        [:refrisk/app-db-delta d])
+      [:refrisk/app-db db])))
+
 (defn send-app-db []
-  ;; TODO send diff
-  (@chsk-send!* [:refrisk/app-db (let [db @(subscribe [::db])]
-                                   (if @pre-send* (@pre-send* db) db))])
-  (when (nil? @id-handler-timer*)
-    (reset! id-handler-timer* (js/setInterval re-frame-sub 1000))))
+  (let [db (get-db)]
+    (when @send-state
+      (send (app-db-msg db)))
+    (when (nil? @id-handler-timer*)
+      (reset! id-handler-timer* (js/setInterval re-frame-sub 1000)))))
 
 (defn start-socket [host]
-  (let [{:keys [send-fn]}
+  (let [{:keys [send-fn ch-recv]}
         (sente/make-channel-socket-client!
           "/chsk"
           {:type     :auto
            :host     host
            :protocol :http
+           :params   {:kind :re-frisk-remote}
            :packer   (sente-transit/get-transit-packer
                        :json
                        {:handlerForForeign (fn [x h] (transit/write-handler (fn [o] "ForeignType")
                                                                             (fn [o] "")))}
                        {})})]
+    (reset! ch-chsk ch-recv)
     (reset! chsk-send!* send-fn)))
 
+(defmulti -event-msg-handler "Sente `event-msg`s handler" :id)
+
+(defmethod -event-msg-handler :chsk/state
+  [{[{was-open? :open?} {now-open? :open?}] :?data :as msg}]
+  (when (and (not was-open?) now-open?)
```

Comment:
so if state is changed we should reset send-state? `(not= was-open? now-open?)` ?